code/control/DNRoot.php

```diff
@@ -1370,10 +1370,11 @@ public function deploySummary(SS_HTTPRequest $request) {
 		);
 		$data = $strategy->toArray();
 
-		// Add in a URL for comparing from->to code changes.
+		// Add in a URL for comparing from->to code changes. Ensure that we have
+		// two proper 40 character SHAs, otherwise we can't show the compare link.
```

Comment:
No, they have to be full SHAs. One of the from values can be "-" (which is the case for new environments)